utils/frame/remote-externalities/src/lib.rs

```diff
@@ -117,36 +117,47 @@ pub struct OfflineConfig {
 pub enum Transport {
 	/// Use the `URI` to open a new WebSocket connection.
 	Uri(String),
-	/// Use existing WebSocket connection.
-	RemoteClient(Arc<WsClient>),
+	/// Use HTTP connection.
+	RemoteClient(Arc<HttpClient>),
 }
 
 impl Transport {
-	fn as_client(&self) -> Option<&WsClient> {
+	fn as_client(&self) -> Option<&HttpClient> {
 		match self {
 			Self::RemoteClient(client) => Some(client),
 			_ => None,
 		}
 	}
 
-	fn as_client_cloned(&self) -> Option<Arc<WsClient>> {
+	fn as_client_cloned(&self) -> Option<Arc<HttpClient>> {
 		match self {
 			Self::RemoteClient(client) => Some(client.clone()),
 			_ => None,
 		}
 	}
 
-	// Open a new WebSocket connection if it's not connected.
+	// Build an HttpClient from a URI.
 	async fn map_uri(&mut self) -> Result<(), &'static str> {
 		if let Self::Uri(uri) = self {
 			log::debug!(target: LOG_TARGET, "initializing remote client to {:?}", uri);
 
-			let ws_client = ws_client(uri).await.map_err(|e| {
+			// If we have a ws uri, try to convert it to an http uri.
+			// We use an HTTP client rather than WS because WS starts to choke with "accumulated
+			// message length exceeds maximum" errors after processing ~10k keys when fetching
+			// from a node running a default configuration.
+			let uri = if uri.starts_with("ws://") {
+				uri.replace("ws://", "http://")
+			} else if uri.starts_with("wss://") {
+				uri.replace("wss://", "https://")
+			} else {
+				uri.clone()
+			};
+			let http_client = HttpClientBuilder::default().build(uri).map_err(|e| {
```

Comment:
Yeah, the websocket client is faster and this is used for instance by the staking miner v1 as well.

Reply:
Inserting data into storage is very cpu heavy so I think it's worth it having a dedicated core working on that.

Otherwise I agree that there's likely little to no improvement adding more threads to do network io. Switching to `FuturesUnordered` for the network io tasks seems like a great idea we could make as part of a future refactor.